src/MoBi.Presentation/Presenter/EditAssignmentBuilderPresenter.cs

```diff
@@ -112,6 +115,16 @@ public void SelectPath()
          setObjectPath(objectPath);
       }
 
+      private ICache<IObjectBase, string> getForbiddenAssignees()
+      {
+         var cache = new Cache<IObjectBase, string>();
+         if (_eventAssignmentBuilder.UseAsValue || _eventAssignmentBuilder.Formula == null)
+            return cache;
+
+         _eventAssignmentBuilder.Formula.ObjectPaths.Select(x => x.TryResolve<IUsingFormula>(_eventAssignmentBuilder)).Each(x => cache.Add(x, AppConstants.Captions.AssigningFormulaCreatesCircularReference));
+         return cache;
+      }
+
```

Comment:
seems to be the same function as in **EditEventBuilderPresenter** - could be defined e.g. as extension method of **EventAssignmentBuilder**?